src/server.ts

```diff
@@ -1,47 +1,98 @@
+    this.pod = pod;
+    this.port = port;
+    this.watcher = new Watcher(pod, this);
+    this.app = undefined;
+    this.httpServer = undefined;
+  }
 
-export function createApp(pod: Pod) {
-  const app = express();
-  app.disable('x-powered-by');
-  pod.plugins.trigger('createServer', app);
-  app.all('/*', async (req: express.Request, res: express.Response) => {
-    try {
-      const route = pod.router.resolve(req.path);
-      if (!route) {
-        res
-          .status(404)
-          .sendFile(fsPath.join(__dirname, './static/', 'error-no-route.html'));
-        return;
+  /**
+   * Returns the `Express` server application.
+   */
+  createApp() {
+    const app = express();
+    this.pod.plugins.trigger('createServer', app);
+    app.disable('x-powered-by');
+    app.all('/*', async (req: express.Request, res: express.Response) => {
+      try {
+        const route = this.pod.router.resolve(req.path);
+        if (!route) {
+          res
+            .status(404)
+            .sendFile(
+              fsPath.join(__dirname, './static/', 'error-no-route.html')
+            );
+          return;
+        }
+        if (route.provider.type === 'static_dir') {
+          res.sendFile(
+            this.pod.getAbsoluteFilePath(
+              (route as StaticRoute).staticFile.podPath
+            )
+          );
+          return;
+        } else {
+          const content = await route.build();
+          res.set('Content-Type', route.contentType);
+          res.send(content);
+        }
+      } catch (err) {
+        nunjucks.configure(fsPath.join(__dirname, './static/'), {
+          autoescape: true,
+          express: app,
+        });
+        res.status(500).render('error-generic.njk', {
+          error: err,
+        });
       }
-      if (route.provider.type === 'static_dir') {
-        res.sendFile(
-          pod.getAbsoluteFilePath((route as StaticRoute).staticFile.podPath)
+    });
+    return app;
+  }
+
+  /**
+   * Starts the web server.
+   */
+  start(startWatcher = false) {
+    this.app = this.createApp();
+    this.httpServer = this.app.listen(this.port, () => {
+      // Only output first time server is started.
+      if (startWatcher) {
+        console.log('   Pod:'.green, `${this.pod.root}`);
+        console.log(
+          'Server:'.green,
+          `${this.pod.env.scheme}://${this.pod.env.host}:${this.port}/`
         );
-        return;
-      } else {
-        const content = await route.build();
-        res.set('Content-Type', route.contentType);
-        res.send(content);
+        console.log(' Ready. Press ctrl+c to quit.'.green);
       }
-    } catch (err) {
-      nunjucks.configure(fsPath.join(__dirname, './static/'), {
-        autoescape: true,
-        express: app,
-      });
-      res.status(500).render('error-generic.njk', {
-        error: err,
-      });
-    }
-  });
-  return app;
+    });
+    startWatcher && this.watcher.start();
```

Comment:
can you confirm none of this is being run when running `ama build` ?

Reply:
Yea, this is just for the server. The `ama build` just invokes what's in `commands/build.ts` and none of the server code is ever invoked.